analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantArgument.cs

```diff
@@ -34,34 +36,26 @@ public sealed class RedundantArgument : SonarDiagnosticAnalyzer
             context.RegisterNodeAction(
                 c =>
                 {
-                    // Can't use optional arguments in expression trees (CS0584), so skip those
-                    if (c.IsInExpressionTree())
+                    if (!c.IsInExpressionTree() // Can't use optional arguments in expression trees (CS0584), so skip those
```

Comment:
We are accessing the SemanticModel right away, I suggest moving `c.IsInExpressionTree()` after the ArgumentList syntax check.